File: src/main/java/br/com/jiratorio/repository/HolidayRepository.java
package br.com.jiratorio.repository;

import br.com.jiratorio.domain.Holiday;
import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface HolidayRepository extends CrudRepository<Holiday, Long> {

    List<Holiday> findAllByBoardId(Long id);

    Page<Holiday> findAllByBoardId(Long id, Pageable pageable);

}

File: src/main/kotlin/br/com/jiratorio/repository/HolidayRepository.kt
package br.com.jiratorio.repository

import br.com.jiratorio.domain.entity.Board
import br.com.jiratorio.domain.entity.Holiday

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.time.LocalDate

@Repository
interface HolidayRepository : CrudRepository<Holiday, Long> {

    fun findAllByBoardId(id: Long): List<Holiday>

    fun findAllByBoardId(id: Long, pageable: Pageable): Page<Holiday>

    fun findAllByBoard(board: Board): List<Holiday>

    fun findByDateAndBoardId(date: LocalDate, boardId: Long): Holiday?

}
